s-modal: Fix keyboard focus on complex views

bnickel opened the pull request:
Right now, we emit show/hide events on `this.element` but listen for them on `this.modalTarget`.  If those two properties aren't the same element, we fail to bring focus into the modal or return focus when hiding the modal (and probably other things).

This fixes it by emitting the event from `modalTarget`.  We could alternatively fix it by binding the events on `element`, but if we made this mistake it's only natural to think consumers would too.

bnickel committed:
Emit to match listeners

Right now, we emit events on `this.element` but listen on `this.modalTarget`.  If they aren't the same element, we fail to do a variety of things.

b-kelly reviewed (changes requested):
I'm ok with changing the emitter to the modal target. This should not be a breaking change for those listening to the controller element as it is either A) the same element or B) a parent of the modal target and will receive the event when it bubbles.

My only requested change would be to update the docs to match. See the `Events` table [here](https://stackoverflow.design/product/components/modals/#events).

dancormier committed:
update docs to reflect new modal target emitter

Per request here https://github.com/StackExchange/Stacks/pull/651#pullrequestreview-670322402

Is this correct? 🤷‍♂️ Am I committing it? 👍
Please correct me if I'm wrong.
